app/src/pages/inside/common/suiteTestToolbar/actionPanel/actionPanel.jsx

```diff
@@ -257,26 +257,72 @@ export class ActionPanel extends Component {
       level,
     } = this.props;
     return (
-      <div className={cx('action-panel', { 'right-buttons-only': !showBreadcrumbs && !hasErrors })}>
-        {showBreadcrumbs && (
-          <Breadcrumbs
-            togglerEventInfo={level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.PLUS_MINUS_BREADCRUMB : {}}
-            breadcrumbEventInfo={
-              level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.ITEM_NAME_BREADCRUMB_CLICK : {}
-            }
-            allEventClick={level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.ALL_LABEL_BREADCRUMB : {}}
-            descriptors={breadcrumbs}
-            onRestorePath={restorePath}
-          />
-        )}
-        {hasErrors && (
-          <GhostButton disabled={!hasValidItems} onClick={onProceedValidItems}>
-            {intl.formatMessage(messages.proceedButton)}
-          </GhostButton>
-        )}
-        <div className={cx('action-buttons')}>
+      <Fragment>
+        <div
+          className={cx('action-panel', { 'right-buttons-only': !showBreadcrumbs && !hasErrors })}
+        >
+          {showBreadcrumbs && (
+            <Breadcrumbs
+              togglerEventInfo={
+                level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.PLUS_MINUS_BREADCRUMB : {}
+              }
+              breadcrumbEventInfo={
+                level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.ITEM_NAME_BREADCRUMB_CLICK : {}
+              }
+              allEventClick={level !== LEVEL_STEP ? SUITES_PAGE_EVENTS.ALL_LABEL_BREADCRUMB : {}}
+              descriptors={breadcrumbs}
+              onRestorePath={restorePath}
+            />
+          )}
+          {hasErrors && (
+            <GhostButton disabled={!hasValidItems} onClick={onProceedValidItems}>
+              {intl.formatMessage(messages.proceedButton)}
+            </GhostButton>
+          )}
+          <div className={cx('action-buttons', 'hidden--l')}>
+            {this.checkVisibility([LEVEL_STEP]) && (
+              <div className={cx('action-button', 'mobile-hidden')}>
+                <GhostMenuButton
+                  title={intl.formatMessage(messages.actionsBtn)}
+                  items={this.actionDescriptors}
+                  disabled={!selectedItems.length}
+                />
+              </div>
+            )}
+            {this.checkVisibility([LEVEL_SUITE, LEVEL_TEST]) && (
```

Comment:
Are you sure we need to duplicate our JSX?
Maybe there is an approach without such duplicates and increase of the content in the files?

Reply:
Did you have something in mind? I can't think of anything that would make the buttons container break under the breadcrumbs (like it would be outside the parent container).

Reply:
What's the problem with adding `flex-direction: column` to the `.action-panel` under media query?
Could you please clarify the layout you want to achieve?
As far as I remember, we decided to move buttons under the breadcrumbs and move them to the right on smaller screens. Is that correct?
If yes, it can be done in the following way:
```
.action-panel {
  /* ... */
  @media(max-width: /* some size */) {
    flex-direction: column;
    align-items: flex-start;
  }
}

.action-buttons {
  /* ... */
  @media(max-width: /* some size */) {
      align-self: flex-end;
      /* other stuff like paddings goes here */
  }
}
```

Reply:
Good idea, I tried it and this works as well. Thanks for the suggestion!